File: CSES/python3/Dynamic Programming/Dice Combinations.py
def count_ways_to_construct_sum(n):
    MOD = 10**9 + 7

    # Initialize a list to store the number of ways for each sum from 0 to n
    ways = [0] * (n + 1)

    # There is 1 way to get a sum of 0 (by not throwing the dice)
    ways[0] = 1

    # Calculate the number of ways for each sum from 1 to n
    for i in range(1, n + 1):
        for j in range(1, 7):  # Possible outcomes of throwing a dice (1 to 6)
            if i >= j:
                ways[i] = (ways[i] + ways[i - j]) % MOD

    return ways[n]

# Example usage:
n = int(input())
result = count_ways_to_construct_sum(n)
print(result)

File: CSES/C++/Dynamic Programming/Dice Combinations.cpp
#include <iostream>
#include <vector>

using namespace std;

const int MOD = 1000000007;

int countWaysToConstructSum(int n) {
    vector<int> ways(n + 1, 0);
    ways[0] = 1;

    for (int i = 1; i <= n; ++i) {
        for (int j = 1; j <= 6; ++j) {
            if (i >= j) {
                ways[i] = (ways[i] + ways[i - j]) % MOD;
            }
        }
    }

    return ways[n];
}

int main() {
    int n;
    cin >> n;
    int result = countWaysToConstructSum(n);
    cout << result << endl;
    return 0;
}
